### PR TITLE
Added aad-pod-identity support

### PR DESCRIPTION
**Pull request by sergiomcalzada**

### CODE REVIEW

**Comment by sesispla** [comment]
@sergiomcalzada I found that some of the elements you created via kubectl are: 
* Deployment
* DaemonSet
* ClusterRole
* ClusterRoleBinding
* ServiceAccount

It is true that terraform don't provide (or at least I don't know) a way to create Kubernetes CRDs

Please consider turning all the YAML files into terraform files that can be properly tracked and managed by terraform.

File: aad-pod-identity-rbac.yml [hunk 1]
```diff
@@ -0,0 +1,257 @@
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: aad-pod-id-nmi-service-account
+  namespace: default
```
Comment (sesispla) [inline aad-pod-identity-rbac.yml:5]: Please use this [terraform resource](https://www.terraform.io/docs/providers/kubernetes/r/service_account.html)

File: aad-pod-identity-rbac.yml [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+  verbs: ["get", "list", "watch"]
+- apiGroups: [""]
+  resources: ["secrets"]
+  verbs: ["get"]
+- apiGroups: ["aadpodidentity.k8s.io"]
+  resources: ["azureidentitybindings", "azureidentities", "azurepodidentityexceptions"]
+  verbs: ["get", "list", "watch"]
+- apiGroups: ["aadpodidentity.k8s.io"]
+  resources: ["azureassignedidentities"]
+  verbs: ["get", "list", "watch"]
```
Comment (sesispla) [inline aad-pod-identity-rbac.yml:76]: Please use this [terraform resource](https://www.terraform.io/docs/providers/kubernetes/r/cluster_role.html)

File: aad-pod-identity-rbac.yml [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+  labels:
+    k8s-app: aad-pod-id-nmi-binding
+subjects:
+- kind: ServiceAccount
+  name: aad-pod-id-nmi-service-account
+  namespace: default
+roleRef:
+  kind: ClusterRole
+  name: aad-pod-id-nmi-role
+  apiGroup: rbac.authorization.k8s.io
```
Comment (sesispla) [inline aad-pod-identity-rbac.yml:91]: Please use this [terraform resource](https://www.terraform.io/docs/providers/kubernetes/r/cluster_role_binding.html)

File: aad-pod-identity-rbac.yml [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+        - mountPath: /run/xtables.lock
+          name: iptableslock
+        livenessProbe:
+          httpGet:
+            path: /healthz
+            port: 8080
+          initialDelaySeconds: 10
+          periodSeconds: 5
+      nodeSelector:
+        beta.kubernetes.io/os: linux
```
Comment (sesispla) [inline aad-pod-identity-rbac.yml:160]: Please use this [terraform resource](https://www.terraform.io/docs/providers/kubernetes/r/daemonset.html)

File: aad-pod-identity-rbac.yml [hunk 5]
```diff
@@ -0,0 +1,257 @@
...
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: aad-pod-id-mic-service-account
+  namespace: default
+---
```
Comment (sesispla) [inline aad-pod-identity-rbac.yml:167]: Please use this [terraform resource](https://www.terraform.io/docs/providers/kubernetes/r/service_account.html)

File: aad-pod-identity-rbac.yml [hunk 6]
```diff
@@ -0,0 +1,257 @@
...
+- apiGroups: [""]
+  resources: ["endpoints"]
+  verbs: ["create", "get","update"]
+- apiGroups: ["aadpodidentity.k8s.io"]
+  resources: ["azureidentitybindings", "azureidentities"]
+  verbs: ["get", "list", "watch", "post"]
+- apiGroups: ["aadpodidentity.k8s.io"]
+  resources: ["azureassignedidentities"]
+  verbs: ["*"]
+---
```
Comment (sesispla) [inline aad-pod-identity-rbac.yml:191]: Please use this [terraform resource](https://www.terraform.io/docs/providers/kubernetes/r/cluster_role.html)

File: aad-pod-identity-rbac.yml [hunk 7]
```diff
@@ -0,0 +1,257 @@
...
+    k8s-app: aad-pod-id-mic-binding
+subjects:
+- kind: ServiceAccount
+  name: aad-pod-id-mic-service-account
+  namespace: default
+roleRef:
+  kind: ClusterRole
+  name: aad-pod-id-mic-role
+  apiGroup: rbac.authorization.k8s.io
+---
```
Comment (sesispla) [inline aad-pod-identity-rbac.yml:206]: Please use this [terraform resource](https://www.terraform.io/docs/providers/kubernetes/r/cluster_role_binding.html)

File: aad-pod-identity-rbac.yml [hunk 8]
```diff
@@ -0,0 +1,257 @@
...
+            path: /healthz
+            port: 8080
+          initialDelaySeconds: 10
+          periodSeconds: 5
+      volumes:
+      - name: k8s-azure-file
+        hostPath:
+          path: /etc/kubernetes/azure.json
+      nodeSelector:
+        beta.kubernetes.io/os: linux
```
Comment (sesispla) [inline aad-pod-identity-rbac.yml:257]: Please use this [terraform resource](https://www.terraform.io/docs/providers/kubernetes/r/deployment.html)

File: aad-pod-identity.yml [hunk 9]
```diff
@@ -0,0 +1,177 @@
...
+        hostPath:
+          path: /var/lib/kubelet
+      - name: certificates
+        hostPath:
+          path: /etc/kubernetes/certs
+      - name: k8s-azure-file
+        hostPath:
+          path: /etc/kubernetes/azure.json
+      nodeSelector:
+        beta.kubernetes.io/os: linux
```
Comment (sesispla) [inline aad-pod-identity.yml:177]: Please use this [terraform resource](https://www.terraform.io/docs/providers/kubernetes/r/deployment.html)

File: aad-pod-identity.yml [hunk 10]
```diff
@@ -0,0 +1,177 @@
...
+        - mountPath: /run/xtables.lock
+          name: iptableslock
+        livenessProbe:
+          httpGet:
+            path: /healthz
+            port: 8080
+          initialDelaySeconds: 10
+          periodSeconds: 5
+      nodeSelector:
+        beta.kubernetes.io/os: linux
```
Comment (sesispla) [inline aad-pod-identity.yml:116]: Please use this [terraform resource](https://www.terraform.io/docs/providers/kubernetes/r/daemonset.html)

File: .gitignore [hunk 11]
```diff
@@ -26,4 +26,5 @@ override.tf.json
 # !example_override.tf
 
 # Include tfplan files to ignore the plan output of command: terraform plan -out=tfplan
-# example: *tfplan*
\ No newline at end of file
+# example: *tfplan*
+/vars/secret.tfvars
```
Comment (sesispla) [inline .gitignore:30]: Could you please confirm what this file is?

**Review by sesispla — CHANGES_REQUESTED** [review]
Having a closer look to it, this looks like a terraform module itself :)

**Comment by sergiomcalzada** [comment]
The yml file comes from an [Azure Github](https://github.com/Azure/aad-pod-identity) and I don't want to split them for easy update... but... Checking the releases they also provide a helm package (that is not documented)... Will try it and check

**Comment by sesispla** [comment]
@sergiomcalzada  Using your inputs here, I'm closing this in favour of the newly implemented [cloudcommons/terraform-kubernetes-aad-pod-identity-rbac](https://github.com/cloudcommons/terraform-kubernetes-aad-pod-identity-rbac). 

This will be made available in the generator with cloudcommons/generator-cli#11 but you can start using the [module](https://registry.terraform.io/modules/cloudcommons/aad-pod-identity-rbac/kubernetes/0.1.0) today

**Pull request closed without merging**